vnext/local-cli/runWindows/utils/VSProjectUtils.ps1

```diff
@@ -0,0 +1,153 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Checks that $PackagesConfigPath is to a packages.config file
+# If $ExistingFile, check that the file exists, else that it doesn't exist
+function Check-PackagesConfigXml {
+    param(
+        [string] $PackagesConfigPath,
+        [bool] $ExistingFile = $True
+    )
+
+    Write-Host
+    Write-Host "Checking " -NoNewLine
+    Write-Host -ForegroundColor White $PackagesConfigPath -NoNewLine
+    Write-Host "..." -NoNewLine
+
+    if ((Split-Path -Path $PackagesConfigPath -Leaf) -ne 'packages.config') {
+        Write-Host " error."
+        throw "$PackagesConfigPath is not a packages.config file."
+    }
+
+    if ($ExistingFile -and -not(Test-Path -Path $PackagesConfigPath)) {
+        Write-Host " error."
+        throw "$PackagesConfigPath does not exist."
+    } elseif (-not($ExistingFile) -and (Test-Path -Path $PackagesConfigPath)) {
+        Write-Host " error."
+        throw "$PackagesConfigPath already exists."
+    }
+
+    Write-Host " done."
+}
+
+# Creates a new packages.config file at $PackagesConfigPath
+# If $CheckConfig, check the file doesn't already exist first
+function Create-PackagesConfigXml {
+    param(
+        [string] $PackagesConfigPath,
+        [bool] $CheckConfig = $True
+    )
+
+    if ($CheckConfig) {
+        Check-PackagesConfigXml -PackagesConfigPath $PackagesConfigPath -ExistingFile $False
+    }
+
+    Write-Host
+    Write-Host "Creating " -NoNewLine
+    Write-Host -ForegroundColor White $PackagesConfigPath -NoNewLine
+    Write-Host "..." -NoNewLine
+
+    [xml]$packageXml = '<?xml version="1.0" encoding="utf-8"?>
+                        <packages>
+                        </packages>'
+
+    $packageXml.Save($PackagesConfigPath)
+
+    Write-Host " done."
+}
+
+# Loads the packages.config file at $PackagesConfigPath
+# If $CheckConfig, check the file exists first
+function Load-PackagesConfigXml {
+    param(
+        [string] $PackagesConfigPath,
+        [bool] $CheckConfig = $True
+    )
+
+    if ($CheckConfig) {
+        Check-PackagesConfigXml -PackagesConfigPath $PackagesConfigPath
+    }
+
+    Write-Host
+    Write-Host "Loading " -NoNewLine
+    Write-Host -ForegroundColor White $PackagesConfigPath -NoNewLine
+    Write-Host "..." -NoNewLine
+
+    [xml]$content = Get-Content -Path $PackagesConfigPath
+
+    Write-Host " done."
+
+    $content.packages.package | foreach {
+            Write-Host "Found package " -NoNewLine
+            Write-Host -ForegroundColor White $_.id -NoNewLine
+            Write-Host " version " -NoNewLine
+            Write-Host -ForegroundColor White $_.version -NoNewLine
+            Write-Host "."
+        }
+
+    return $content;
+}
+
+# Update package $PackageName to version $PackageVersion in the packages.config
+# file at $PackagesConfigPath
+# If $CheckConfig, check the file exists first
+function Update-PackagesConfigXml {
```

Comment:
Is the motivation here to allow updating packages.config before build, or is more manual?

I like the idea, but also want us to be careful not to let the Visual Studio path degrade if we do add this to the CLI only.

I.e. if we can figure out a way to manage package updates in build logic that would be a bit preferable. That's likely a bit harder though if VS tries to restore before the build.

Reply:
this script is just a set of atomic actions for dealing with packages.config from the command line. how and when we use them will come later, as I make updates to the cli

Reply:
the way autolinking works is that when you run "run-windows" all the native things get updated so the build works. for us, this will include: generating some targets, some source files, and potentially updating dependencies. everything I'm doing is making sure we use official tools as much as possible so we don't break VS